src/components/datagrid/body/data_grid_cell.tsx

```diff
@@ -298,6 +298,21 @@ export class EuiDataGridCell extends Component<
       this.recalculateLineHeight();
     }
 
+    if (
+      this.props.colIndex === 0 && // once per row
+      this.props.columnId === prevProps.columnId && // if this is still the same column
+      this.props.rowIndex === prevProps.rowIndex && // if this is still the same row
+      this.props.style?.top !== prevProps.style?.top // if the top position has changed
+    ) {
+      const previousTop = parseFloat(prevProps.style?.top as string);
+      const currentTop = parseFloat(this.props.style?.top as string);
+      this.props.rowHeightUtils?.compensateForLayoutShift(
+        this.props.rowIndex,
+        currentTop - previousTop,
+        this.props.rowHeightsOptions?.scrollAnchorRow
+      );
+    }
```

Comment:
[optional microperf] Should we be checking for whether `rowHeightOptions.scrollAnchorRow` is defined first? If it's not, then we don't need to bother with the other conditionals or parsing numbers, correct?

```suggestion
    if (
      this.props.rowHeightsOptions?.scrollAnchorRow &&
      this.props.colIndex === 0 && // once per row
      this.props.columnId === prevProps.columnId && // if this is still the same column
      this.props.rowIndex === prevProps.rowIndex && // if this is still the same row
      this.props.style?.top !== prevProps.style?.top // if the top position has changed
    ) {
      const previousTop = parseFloat(prevProps.style?.top as string);
      const currentTop = parseFloat(this.props.style?.top as string);
      this.props.rowHeightUtils?.compensateForLayoutShift(
        this.props.rowIndex,
        currentTop - previousTop,
        this.props.rowHeightsOptions?.scrollAnchorRow
      );
    }
```

Reply:
I added this in 00accc704bc4a2ea685b426328a2974f5e0f75f1 because it didn't seem to do any harm

Reply:
Awesome, I appreciate it! :heart: